packages/oauth/src/index.ts

```diff
@@ -802,14 +808,16 @@ function detectExpiredOrExpiringTokens(authResult: AuthorizeResult, currentUTCSe
   const tokensToRefresh: string[] = [];
   const EXPIRY_WINDOW: number = 7200; // 2 hours
 
-  if (authResult.botRefreshToken !== undefined && authResult.botTokenExpiresAt !== undefined) {
+  if (authResult.botRefreshToken &&
+    (authResult.botTokenExpiresAt !== undefined && authResult.botTokenExpiresAt !== null)) {
```

Comment:
Although 0 value does not mean anything here, just in case, I avoided truthy value check for expiresAt values